src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/histogram.js

```diff
@@ -0,0 +1,533 @@
+            .attr("transform",d => "translate(" + x(d.x0) + "," + y(d.length) + ")")
+            .attr("width",d => x(d.x1) - x(d.x0))
+            .attr("height",d => canvas.inner.height - y(d.length))
+            .style("fill",accentColor)
+
+        // Create clip path
+        let defs = svg.append('defs')
+        defs.append('clipPath')
+            .attr('id', 'hist-clip-path')
+            .append('rect')
+            .attr('width', canvas.inner.width)
+            .attr('height', canvas.inner.height)
+
+        return [bars,{x:x,y:y,zoom:1.0},{x:xAxis,y:yAxis}];
+    }
+
+    /**
+     * Creates labels on axes if they are defined.
+     */
+    createLabels(axis,svg,canvas) {
+        let fontStyle = "10px DejaVuSansMonoBook";
+        if (axis.x.label !== undefined) {
+            let padding_y = 20;
+            svg.append("text")
+                .attr("text-anchor","end")
+                .attr("style",LABEL_STYLE)
+                .attr("x",canvas.margin.left + (this.getTextWidth(axis.x.label,fontStyle) / 2))
+                .attr("y",canvas.inner.height + canvas.margin.top + padding_y)
+                .text(axis.x.label);
+        }
+
+        if (axis.y.label !== undefined) {
+            let padding_y = 15;
+            svg.append("text")
+                .attr("text-anchor","end")
+                .attr("style",LABEL_STYLE)
+                .attr("transform","rotate(-90)")
+                .attr("y",-canvas.margin.left + padding_y)
+                .attr("x",-canvas.margin.top - (canvas.inner.height/2) + (this.getTextWidth(axis.y.label,fontStyle) / 2))
+                .text(axis.y.label);
+        }
+    }
+
+    /**
+     * Return the text width. Ensures that labels on x axis wont overlap to keeps them readable.
+     */
+    getTextWidth(text,font) {
+        const canvas  = document.createElement("canvas");
+        const context = canvas.getContext("2d");
+        context.font  = font;
+        const metrics = context.measureText("  " + text);
+        return metrics.width;
+    }
+
+    /**
+     * Return the extrema of the data and and paddings that ensure data will fit into the
+     * drawing area.
+     *
+     * It traverses through data getting minimal and maximal values, and calculates padding based on
+     * span calculated from above values, multiplied by 10% so that the plot is a little bit smaller
+     * than the container.
+     */
+    getExtremesAndDeltas(dataPoints) {
+        let xMin = dataPoints[0]
+        let xMax = dataPoints[0]
+
+        dataPoints.forEach(value => {
+            if (value < xMin) { xMin = value }
+            if (value > xMax) { xMax = value }
+        });
+
+        let dx        = xMax - xMin;
+        let padding_x = 0.1 * dx;
+
+        return {xMin:xMin,xMax:xMax,paddingX:padding_x,dx:dx};
+    }
+
+    /**
+     * Return margins for plots drawing area.
+     */
+    getMargins(axis) {
+        if (axis.x.label === undefined && axis.y.label === undefined) {
```

Comment:
VERY mminor thing. Just create `let noXAxis = axis.x.label === undefined` and use it in ifs - it will be easier to read